app/controllers/collaborations_controller.rb

```diff
@@ -0,0 +1,38 @@
+class CollaborationsController < ApplicationController
+  helper_method :mailbox, :conversation
+
+  def create
+    require_login
+    project = Project.find_by(id: params[:project_id])
+    if project && !collaboration_exists?(current_user, project)
+      @pending = Collaboration.create(collaborator_id: current_user.id, project_id: project.id)
+      # recipient = User.find_by(id: project.creator)
+      # conversation = current_user.send_message(recipient, "#{@pending.collaborator.username} has requested to join your project: #{project.title}", 'Request').conversation
+
+      flash[:notice] = "You have requested to collaborate on '#{project.title}'"
+    else
+      flash[:errors] = ["You have already made a request to collaborate on this project."]
```

Comment:
You could reach here if the project is not found - which would give a misleading error message.